pkg/oc/cli/admin/node/logs/logs.go

```diff
@@ -0,0 +1,535 @@
+const LogsRecommendedCommandName = "node-logs"
+
+var (
+	logsLong = templates.LongDesc(`
+		Display and filter node logs
+
+		This command retrieves logs for the node. The default mode is to query the
+		systemd journal on supported operating systems, which allows searching, time
+		based filtering, and unit based filtering. You may also use the --path argument
+		to see a list of log files available under /var/logs/ and view those contents
+		directly.
+
+		Node logs may contain sensitive output and so are limited to privileged node
+		administrators. The system:node-admins role grants this permission by default.
+		You check who has that permission via:
+
+		    $ oc adm policy who-can --all-namespaces get nodes/log
+		`)
+
+	logsExample = templates.Examples(`
+		# Show kubelet logs from all masters
+		%[1]s %[2]s -r master -u kubelet
+	`)
+)
+
+// LogsOptions holds all the necessary options for running oc adm node-logs.
+type LogsOptions struct {
+	Resources []string
+	Selector  string
+	Role      string
+
+	// the log path to fetch
+	Path string
+
+	// --path=journal specific arguments
+	Pattern                string
+	PatternCaseInsensitive bool
+	Units                  []string
+	SinceTime              string
+	UntilTime              string
+	Tail                   int
+	Output                 string
+
+	// output format arguments
+	Raw   bool
+	Unify bool
+
+	RESTClientGetter func(mapping *meta.RESTMapping) (resource.RESTClient, error)
+	Builder          *resource.Builder
+
+	genericclioptions.IOStreams
+}
+
+func NewLogsOptions(streams genericclioptions.IOStreams) *LogsOptions {
+	return &LogsOptions{
+		Path:      "journal",
+		IOStreams: streams,
+	}
+}
+
+// NewCmdLogs creates a new logs command that supports OpenShift resources.
+func NewCmdLogs(name, baseName string, f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewLogsOptions(streams)
+	cmd := &cobra.Command{
+		Use: "node-logs [-l LABELS] [NODE...]",
+		DisableFlagsInUseLine: true,
+		Short:   "Display and filter node logs",
+		Long:    logsLong,
+		Example: fmt.Sprintf(logsExample, baseName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.RunLogs())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Path, "path", o.Path, "Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the journal on supported operating systems.")
+
+	cmd.Flags().StringSliceVarP(&o.Units, "unit", "u", o.Units, "Return log entries from the specified unit(s). Only applies to node journal logs.")
+	cmd.Flags().StringVarP(&o.Pattern, "search", "S", o.Pattern, "Filter log entries by the provided regex pattern. Only applies to node journal logs.")
+	cmd.Flags().BoolVar(&o.PatternCaseInsensitive, "ignore-case", o.PatternCaseInsensitive, "Make the search filter case-insensitive. Only applies to node journal logs.")
```

Comment:
`--case-sensitive`

Reply:
Lol, we were case sensitive before and Maciej suggested ignore-case.

Reply:
@deads2k `ignore-case` is much more broadly used in UNIX CLI.